File: textProcessor/src/main/haskell/Harmonie/Common/PartOfSpeech.hs
{-# LANGUAGE DeriveDataTypeable #-}

module Harmonie.Common.PartOfSpeech where

import Data.Typeable

data PartOfSpeech = Noun | Verb | Adverb | Adjective | Conjunction | Article | Pronoun | Preposition | NonWord
    deriving (Show, Eq, Typeable, Read)

File: common/src/main/kotlin/am5800/common/PartOfSpeech.kt
package am5800.common

enum class PartOfSpeech {
  Verb,
  Noun,
  ProperNoun,
  Article,
  Preposition,
  Other,
  Punctuation,
  ForeignLanguage,
  Conjunction,
  Pronoun,
  Adverb,
  Adjective,
  Digit;

  companion object {
    fun parse(string: String): PartOfSpeech {
      return PartOfSpeech.valueOf(string)
    }
  }
}